### DISCUSSION TITLE
Suggestion: have a setting somewhere that can hold an IP address, and a way to fill that in for any application

### DISCUSSION THREAD

**Discussion by ericblade** [linuxserver/Heimdall#1286] [Ideas]
Like a macro replacement or something, doesn't matter how configurable it is, but I'd like to just be able to set $MY_APP_SERVER = '192.168.0.100' or something and then i can use http://$MY_APP_SERVER:port/url in each of the application buttons.  

I just had to change the address where everything in my system points to, and Heimdall was the only pain point I had in doing that, everything else was a simple find/replace in my configs.

[a comment by an automated account is not shown]

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
Would be nice to be able to use en Environment variable for this. So if one day the address changes, you could just update the domain or IP value once.

**Reply by ericblade** [reply]
Agree

**Comment by KBearMan** [comment]
I've been using Heimdall for a while and I love it. But my server's IP has changed a few times, and this last time when I had 30 containers that all needed to update the IP address for, it was brutal and time consuming. Unfortunately this might be the issue that causes me to stop using heimdall altogether. As I add more containers I'm just not going to be able to go in and manually change every IP address, especially when the process isn't streamlined at the moment for a single container, let alone 30+.

**Comment by j0nnymoe** [comment]
If your server IP is changing, I would argue that you should be statically setting your servers ip rather than letting DHCP randomly change it. Also re addresses within Heimdall, it's better practice to use DNS addresses rather than IP where you can.

**Reply by KBearMan** [reply]
I don't have the ability to statically set my IP for my server running this. I'm not really sure how I would use DNS in my case. I have a single server running several docker containers. Each container is mapped to a different port on the host machine. The ports ARE static, but the machine's IP does change.  So I need a way for heimdall to  understand that everything it's pointing to shares the same IP but it's just mapping to different ports.

**Reply by j0nnymoe** [reply]
Basic ISP provide routers can set static DHCP leases, failing that you can do it directly in your OS.
